connectors/sources/servicenow.py

```diff
@@ -194,6 +217,83 @@ async def filter_services(self):
             logger.exception(f"Error while filtering services. Exception: {exception}.")
             raise
 
+    async def fetch_data(self, url, params):
+        """Hit API call and iterate over the fetched response.
+
+        Args:
+            url (str): ServiceNow query to be executed.
+            params (dict): Params to format the query.
+
+        Yields:
+            dict: Formatted document.
+        """
+
+        try:
+            async for response in self._api_call(url=url, params=params):
+                for record in response:  # pyright: ignore
+                    yield record
+
+        except Exception as exception:
+            logger.warning(
+                f"Skipping data for {url} with {params}. Exception: {exception}."
+            )
+
+    async def fetch_attachment_content(self, metadata, timestamp=None, doit=False):
+        """Fetch attachment content via metadata.
+
+        Args:
+            metadata (dict): Attachment metadata.
+            timestamp (timestamp, None): Attachment last modified timestamp. Defaults to None.
+            doit (bool, False): Whether to get content or not. Defaults to False.
+
+        Returns:
+            dict: Document with id, timestamp & content.
+        """
+
+        attachment_size = int(metadata["size_bytes"])
+        if not (doit and attachment_size > 0):
+            return
+
+        attachment_name = metadata["file_name"]
+        if os.path.splitext(attachment_name)[-1] not in TIKA_SUPPORTED_FILETYPES:
+            logger.warning(f"{attachment_name} is not supported by TIKA, skipping.")
+            return
```

Comment:
```suggestion
        attachment_extension = os.path.splitext(attachment_name)[-1]
        if  attachment_extension not in TIKA_SUPPORTED_FILETYPES:
            logger.warning(f"Files with the extension {attachment_extension} are not supported by TIKA, skipping {attachment_name}.")
            return
```

suggestion: I would use a variable here and also indicate in the log message, that the file type is not supported.